### PR TITLE
[commabody] Add new body teleop ui

### PR DESCRIPTION
**Pull request by nuwandavek**
- This needs https://github.com/commaai/openpilot/pull/29087 and https://github.com/commaai/openpilot/pull/29118 merged
- This also needs the packages installed

ToDo:
- [x] Make this PR testable without the other two PRs
- [x] Test on wheeled body
- [x] Test on non-wheeled body
---
 ~Tested and works fine. Can be merged after figuring out the packages and livestream encoder stuff~
 
 - [x] Split out all the dependencies
 - [x] Split out the js files

### CODE REVIEW

**Comment by nuwandavek** [comment]
@adeebshihadeh The build fails because these packages are not in agnos? Can we merge this somehow or do we need to wait till we have it in the next agnos?

**Comment by adeebshihadeh** [comment]
Which packages? I got all the requested ones in AGNOS 8.

**Comment by nuwandavek** [comment]
> Which packages? I got all the requested ones in AGNOS 8.

It was just not in poetry. All tests pass now!

**Review by adeebshihadeh — COMMENTED** [review]
Can you split out the new dependencies?

**Comment by nuwandavek** [comment]
> Can you split out the new dependencies?

Split out as in? Add them at the end? They're all under main now.

**Comment by adeebshihadeh** [comment]
No, they should go in their own PR.

File: tools/bodyteleop/static/js/controls.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const keyVals = {w: 0, a: 0, s: 0, d: 0}
+
+export function getXY(){
+    let x = -keyVals.w + keyVals.s
+    let y = -keyVals.d + keyVals.a
+    return {x, y}
+}
+
+export const handleKeyX = (key, setValue) => {
+    if (['w', 'a', 's', 'd'].includes(key)){
```
Comment (adeebshihadeh) [inline tools/bodyteleop/static/js/controls.js:10]: two spaces for indentation

File: tools/bodyteleop/static/index.html [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                        <div class="form-check form-switch">
+                            <input class="form-check-input" type="checkbox" role="switch" id="find-person" disabled>
+                            <label class="form-check-label small" for="find-person">Find People!</label>
+                        </div>                          
+                    </div>
+                </div>
+            </div> -->
+        </div>
+        <script src="/static/js/jsmain.js" type="module"></script>
+    </body>
```
Comment (adeebshihadeh) [inline tools/bodyteleop/static/index.html:119]: same here

File: tools/bodyteleop/bodyav.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  kind = "video"
+
+  _start: float
+  _timestamp: int
+
+  def __init__(self):
+    super().__init__()
+    sock_name = 'livestreamDriverEncodeData'
+    # os.environ['ZMQ'] = '1'
+    # addr='192.168.1.101'
```
Comment (adeebshihadeh) [inline tools/bodyteleop/bodyav.py:41]: this dead?

File: tools/bodyteleop/bodyav.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    frame = self.codec.decode(packet)[0]
+    frame.pts = self.audio_samples
+    self.audio_samples += frame.samples
+    self.chunk_number = self.chunk_number + 1
+    return frame
+
+
+async def play_sound(sound):
+  chunk = 5120
+  print("playing", sound)
```
Comment (adeebshihadeh) [inline tools/bodyteleop/bodyav.py:144]: remove print or use proper logging

File: tools/bodyteleop/static/js/webrtc.js [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    };
+
+    pc = new RTCPeerConnection(config);
+
+    // connect audio / video
+    pc.addEventListener('track', function(evt) {
+        console.log("Adding Tracks!")
+        if (evt.track.kind == 'video')
+            document.getElementById('video').srcObject = evt.streams[0];
+        else
```
Comment (adeebshihadeh) [inline tools/bodyteleop/static/js/webrtc.js:20]: use consistent bracket style

File: tools/bodyteleop/static/js/webrtc.js [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
...)|hs\-c|ht(c(\-| |_|a|g|p|s|t)|tp)|hu(aw|tc)|i\-(20|go|ma)|i230|iac( |\-|\/)|ibro|idea|ig01|ikom|im1k|inno|ipaq|iris|ja(t|v)a|jbro|jemu|jigs|kddi|keji|kgt( |\/)|klon|kpt |kwc\-|kyo(c|k)|le(no|xi)|lg( g|\/(k|l|u)|50|54|\-[a-w])|libw|lynx|m1\-w|m3ga|m50\/|ma(te|ui|xo)|mc(01|21|ca)|m\-cr|me(rc|ri)|mi(o8|oa|ts)|mmef|mo(01|02|bi|de|do|t(\-| |o|v)|zz)|mt(50|p1|v )|mwbp|mywa|n10[0-2]|n20[2-3]|n30(0|2)|n50(0|2|5)|n7(0(0|1)|10)|ne((c|m)\-|on|tf|wf|wg|wt)|nok(6|i)|nzph|o2im|op(ti|wv)|oran|owg1|p800|pan(a|d|t)|pdxg|pg(13|\-([1-8]|c))|phil|pire|pl(ay|uc)|pn\-2|po(ck|rt|se)|prox|psio|pt\-g|qa\-a|qc(07|12|21|32|60|\-[2-7]|i\-)|qtek|r380|r600|raks|rim9|ro(ve|zo)|s55\/|sa(ge|ma|mm|ms|ny|va)|sc(01|h\-|oo|p\-)|sdk\/|se(c(\-|0|1)|47|mc|nd|ri)|sgh\-|shar|sie(\-|m)|sk\-0|sl(45|id)|sm(al|ar|b3|it|t5)|so(ft|ny)|sp(01|h\-|v\-|v )|sy(01|mb)|t2(18|50)|t6(00|10|18)|ta(gt|lk)|tcl\-|tdg\-|tel(i|m)|tim\-|t\-mo|to(pl|sh)|ts(70|m\-|m3|m5)|tx\-9|up(\.b|g1|si)|utst|v400|v750|veri|vi(rg|te)|vk(40|5[0-3]|\-v)|vm40|voda|vulc|vx(52|53|60|61|70|80|81|83|85|98)|w3c(\-| )|webc|whit|wi(g |nc|nw)|wmlb|wonu|x700|yas\-|your|zeto|zte\-/i.test(a.substr(0,4))) check = true;})(navigator.userAgent||navigator.vendor||window.opera);
```
Comment (adeebshihadeh) [inline tools/bodyteleop/static/js/webrtc.js:67]: what's this?

File: tools/bodyteleop/static/main.css [hunk 7]
```diff
@@ -0,0 +1,189 @@
+body{
+    background: #333 !important;
+    color: #fff !important;
+    display: flex;
+    justify-content: center;
+    align-items: start;
+}
+
+p{
```
Comment (adeebshihadeh) [inline tools/bodyteleop/static/main.css:9]: space before bracket

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
This is pretty big and the style is all over the place. Can you delete any dead code and make the style consistent?

File: tools/bodyteleop/static/js/webrtc.js (hunk 6 above)
Comment (nuwandavek) [reply tools/bodyteleop/static/js/webrtc.js:71]: It is to check if it is mobile or not. There are some differences in webrtc conditions for chrome on desktop and mobile phones. This is the most comprehensive way to check that I found.

**Comment by nuwandavek** [comment]
Fixed a bunch of styling issues!

**Comment by haraschax** [comment]
LGTM

File: tools/bodyteleop/web.py [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+os.environ['OPENBLAS_NUM_THREADS'] = '1'
+os.environ['MKL_NUM_THREADS'] = '1'
+
+logger = logging.getLogger("pc")
+logging.basicConfig(level=logging.INFO)
+
+pcs = set()
+
+pm = messaging.PubMaster(['testJoystick'])
+sm = messaging.SubMaster(['carState', 'logMessage'])
```
Comment (adeebshihadeh) [inline tools/bodyteleop/web.py:25]: should be gated on `if __name__ == "__main__"`

File: tools/bodyteleop/web.py [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+
+from common.basedir import BASEDIR
+from aiohttp import web
+from aiortc import RTCPeerConnection, RTCSessionDescription
+
+import cereal.messaging as messaging
+from tools.bodyteleop.bodyav import BodyMic, WebClientSpeaker, force_codec, play_sound, MediaBlackhole, EncodedBodyVideo
+
+os.environ['OPENBLAS_NUM_THREADS'] = '1'
+os.environ['MKL_NUM_THREADS'] = '1'
```
Comment (adeebshihadeh) [inline tools/bodyteleop/web.py:17]: this is done in launch_env.sh

**Pull request merged**